pkg/sensors/program/genericoverride.go

```diff
@@ -59,24 +80,24 @@ func createKProbeOverrideProg(attachFunc string) *Program {
 	overrideProg.PinPath = path.Join("__override__", "kprobe", attachFunc)
 	overrideProg.unloaderOverride = &unloader.CustomUnloader{
 		UnloadFunc: func(_ bool) error {
-			deleteOverrideProg(OverrideTypeKProbe, attachFunc)
-			return nil
+			return deleteOverrideProg(OverrideTypeKProbe, attachFunc, &ret)
 		},
 	}
 
 	overrideTasksMap := MapBuilder("override_tasks", overrideProg)
 	overrideTasksMap.PinPath = path.Join("__override__", "override_tasks")
 	overrideTasksMap.SetMaxEntries(OverrideMapMaxEntries)
 
-	return overrideProg
+	ret.prog = overrideProg
+	return &ret
 }
 
 func GetOverrideProg(overrideType OverrideType, attachFunc string) (*Program, *Map) {
-	var overrideProg *Program
+	var overrideProg *genericOverride
 	var ok bool
 
 	if overrideProgMap == nil {
-		overrideProgMap = make(map[string]*Program)
+		overrideProgMap = make(map[string]*genericOverride)
```

Comment:
this needs more explanation on why is this needed.. please use commit log message to describe what you are doing and why.. for now and for future ;-)